src/brogue/Items.c

```diff
@@ -3387,7 +3387,10 @@ short getLineCoordinates(short listOfCoordinates[][2], const short originLoc[2],
             boolean isMonster = monst
                 && !(monst->bookkeepingFlags & MB_SUBMERGED)
                 && !monsterIsHidden(monst, monsterAtLoc(originLoc[0], originLoc[1]));
-            boolean isAllyOfCaster = ((monst && monst->creatureState == MONSTER_ALLY) == isCastByPlayer);
+            creature *caster = monsterAtLoc(originLoc[0], originLoc[1]);
+            boolean isAllyOfCaster = // true if monster and caster are on the same side
+                (monst && caster && monstersAreTeammates(monst, caster)) ||
+                (monst && !caster && !monstersAreTeammates(monst, &player)); // a creature that isn't on our side is on turrets' side
```

Comment:
I'm confused by this, why are turrets a special case here? They are monsters - does the bolt not fire from the monster cell or something?

Reply:
You are right. I was making a wrong assumption. Should be good now.